src/Models/DataImport.php

```diff
@@ -67,6 +67,13 @@ public function delete()
         parent::delete();
     }
 
+    public function cancel()
+    {
+        throw_if($this->status >= Statuses::Finalized, DataImportException::cannotBeCanceled());
```

Comment:
I dont' know how this will age. Lets be more specific (in_array or even cancellable method in the Stastuses enum)